src/com/example/ui.cljs

```diff
@@ -30,24 +33,63 @@
 
 (def ui-header (comp/factory Header))
 
+(defn field [{:keys [label valid? error-message input-class id] :as props}]
+  (let [input-props (-> props
+                        (dissoc :label :valid? :error-message :input-class :options))]
+    (div
+     (dom/label {:htmlFor label} label)
+     (input-class input-props)
+     (dom/div {:classes [(when valid? "hidden")]}
+              error-message))))
+
+(defsc Question [this {:question/keys [id label options]}]
+  {:query [:question/id
+           :question/label
+           :ui/selected
+           {:question/options [:option/label
+                               :option/value]}]
+   :ident (fn [] [:question/id id])
+   :initial-state {}}
+  (div :.mb-4
+       (h2 :.text-lg label)
+       (map (fn [o]
+              (field {:input-class  input
+                      :name         id
+                      :label        (:option/label o)
+                      :value        (str (:option/value o))
+                      :id           (:option/label o)
+                      :key          (:option/value o)
+                      :type         "radio"
+                      :autoComplete "off"
+                      :onChange     #(m/set-integer! this :ui/selected :event %)}))
```

Comment:
when this field changes, we set the `ui/selected` field to the value that comes from the event object. `m/set-integer!` does the coercion to an int for us